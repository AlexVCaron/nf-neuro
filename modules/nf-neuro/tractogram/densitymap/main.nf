process TRACTOGRAM_DENSITYMAP {
    tag "$meta.id"
    label 'process_single'

    container "scilus/scilpy:2.2.0_cpu"

    input:
    tuple val(meta), path(tractogram)

    output:
    tuple val(meta), path("*__*.nii.gz"), emit: density_map
    path "versions.yml"                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def binary = task.ext.is_binary ? "--binary" : ""
    def endpoints_only = task.ext.endpoint_only ? "--endpoints_only" : ""

    """
    bname=\$(basename ${tractogram} .trk)
    scil_tractogram_compute_density_map $tractogram ${prefix}__\${bname}.nii.gz ${binary} ${endpoints_only}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    scil_tractogram_compute_density_map -h

    touch ${prefix}__AF_L.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: \$(uv pip -q -n list | grep scilpy | tr -s ' ' | cut -d' ' -f2)
    END_VERSIONS
    """
}
