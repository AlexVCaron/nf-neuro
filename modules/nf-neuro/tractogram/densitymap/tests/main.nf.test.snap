{
    "densitymap - map": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,a9cc8ba016a680ec8cbac940c45e5464"
                    ]
                ],
                "1": [
                    "versions.yml:md5,99078ff3904c269f687e76f248685e2d"
                ],
                "density_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,a9cc8ba016a680ec8cbac940c45e5464"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,99078ff3904c269f687e76f248685e2d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-23T00:30:48.604554342"
    },
    "densitymap - mask": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,5c682d295c593e132295e21e753a6f4f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,99078ff3904c269f687e76f248685e2d"
                ],
                "density_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,5c682d295c593e132295e21e753a6f4f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,99078ff3904c269f687e76f248685e2d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-23T00:30:59.827759348"
    },
    "densitymap - stub-run": {
        "content": [
            [
                "versions.yml:md5,99078ff3904c269f687e76f248685e2d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-23T00:31:15.408992193"
    }
}